packages/odf/components/kms-config/name-address-port.tsx

```diff
@@ -63,23 +70,31 @@ export const NameAddrPort: React.FC<NameAddrPortProps> = ({
         fieldId="kms-service-name"
         label={t('Connection name')}
         className={`${className}__form-body`}
-        helperTextInvalid={validateConnectionName(kmsState.name.value, t)}
-        validated={isValid(kmsState.name.valid)}
-        helperText={t(
-          'An unique name for the key management service within the project. Name must only include alphanumeric characters, "-", "_" or "."'
-        )}
         isRequired
       >
         <TextInput
           value={kmsState.name.value}
-          onChange={setServiceName}
+          onChange={(_event, name: string) => setServiceName(name)}
           type="text"
           id="kms-service-name"
           name="kms-service-name"
           isRequired
           validated={isValid(kmsState.name.valid)}
           data-test="kms-service-name-text"
         />
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isValid(kmsState.name.valid) ? 'default' : 'error'}
```

Comment:
```suggestion
              variant={getValidatedProp(!kmsState.name.valid)}
```

I think isValid itself returns 'default' | 'error'